huaweicloud/services/dns/resource_huaweicloud_dns_recordset.go

```diff
@@ -0,0 +1,482 @@
+		KeepResponseBody: true,
+		OkCodes: []int{
+			202,
+		},
+	}
+	updateDNSRecordsetStatusOpt.JSONBody = utils.RemoveNil(buildUpdateDNSRecordsetStatusBodyParams(d))
+	_, err := recordsetClient.Request("PUT", updateDNSRecordsetStatusPath, &updateDNSRecordsetStatusOpt)
+	if err != nil {
+		return fmt.Errorf("error updating DNS recordset status: %s", err)
+	}
+	return nil
+}
+
+func buildUpdateDNSRecordsetBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        utils.ValueIngoreEmpty(d.Get("name")),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
+		"type":        utils.ValueIngoreEmpty(d.Get("type")),
+		"ttl":         utils.ValueIngoreEmpty(d.Get("ttl")),
+		"records":     utils.ValueIngoreEmpty(d.Get("records")),
+		"weight":      utils.ValueIngoreEmpty(d.Get("weight")),
+	}
+	return bodyParams
+}
+
+func buildUpdateDNSRecordsetStatusBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"status": utils.ValueIngoreEmpty(d.Get("status")),
+	}
+	return bodyParams
+}
+
+func resourceDNSRecordsetDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteDNSRecordset: Delete DNS recordset
+	var (
+		deleteDNSRecordsetHttpUrl = "v2.1/zones/{zone_id}/recordsets/{recordset_id}"
+		deleteDNSRecordsetProduct = "dns"
+	)
+	deleteDNSRecordsetClient, err := cfg.NewServiceClient(deleteDNSRecordsetProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DNS Client: %s", err)
+	}
+
+	zoneID := d.Get("zone_id").(string)
+	deleteDNSRecordsetPath := deleteDNSRecordsetClient.Endpoint + deleteDNSRecordsetHttpUrl
+	deleteDNSRecordsetPath = strings.ReplaceAll(deleteDNSRecordsetPath, "{zone_id}", zoneID)
+	deleteDNSRecordsetPath = strings.ReplaceAll(deleteDNSRecordsetPath, "{recordset_id}", d.Id())
+
+	deleteDNSRecordsetOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			202,
+		},
+	}
+	_, err = deleteDNSRecordsetClient.Request("DELETE", deleteDNSRecordsetPath, &deleteDNSRecordsetOpt)
+	if err != nil {
+		return diag.Errorf("error deleting DNS recordset: %s", err)
+	}
+
+	if err := waitForDNSRecordsetDeleted(ctx, deleteDNSRecordsetClient, d); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func waitForDNSRecordsetDeleted(ctx context.Context, recordsetClient *golangsdk.ServiceClient,
+	d *schema.ResourceData) error {
+	stateConf := &resource.StateChangeConf{
+		Target:       []string{"DELETED"},
+		Pending:      []string{"ACTIVE", "PENDING", "ERROR"},
+		Refresh:      dnsRecordsetStatusRefreshFunc(d, recordsetClient),
+		Timeout:      10 * time.Minute,
+		Delay:        5 * time.Second,
+		PollInterval: 5 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for DNS recordset (%s) to become DELETED: %s", d.Id(), err)
```

Comment:
It is more appropriate to modify "become" to "be"

Reply:
OK